## Task 10 — Revenue by Subscription Plan

### Business Question

Which subscription plans generate the highest total monthly revenue?

### Analysis

Group customers by subscription plan, calculate the total monthly revenue and average subscription fee for each plan, and sort the results by total monthly revenue in descending order.

In [1]:
import pandas as pd
import numpy as np
customers = pd.read_csv("starlink_customers.csv")
subscriptions = pd.read_csv("subscriptions.csv")
starlink = customers.merge(
    subscriptions,
    left_on = "unique_id",
    right_on = "customer_id",
    how = "left"
)
revenue_by_plan = starlink.groupby("plan_name").agg(
    total_revenue = ("monthly_fee","sum"),
    mean = ("monthly_fee","mean")
).sort_values(
    "total_revenue",
    ascending = False
)
print(revenue_by_plan)

             total_revenue   mean
plan_name                        
Residential       715080.0  120.0
Business          254750.0  250.0
Roam              166240.0   80.0
Mini               47200.0   50.0


### Result

The Residential plan generates the highest total monthly revenue at 715,080, followed by Business at 254,750 and Roam at 166,240. Mini generates the lowest total monthly revenue at 47,200. The average subscription fee is highest for Business at 250, followed by Residential at 120, Roam at 80, and Mini at 50.

### Business Insight

Residential is the main contributor to total monthly revenue because it has the largest customer base, while Business generates the highest average revenue per customer. This shows that overall revenue is driven mainly by customer volume, whereas Business has a higher revenue contribution per individual customer.